### ISSUE TITLE
Compatibility issue with diff-hl

### ISSUE THREAD

**Issue by ginqi7** [emacs-sideline/sideline#10]
I meet a strange issue, when using [diff-hl](https://github.com/dgutov/diff-hl) and sideline together.

When both diff-hl and sideline are displayed on the same empty line, the sideline content will occupy two virtual lines.
There are tree conditions:
1. Both diff-hl-mode and sideline-mode are enabled.
2. diff-hl-mode fringe and sideline-mode overlay are displayed on a same line.
3. the line is empty. (If there are non-empty strings on the line, it's not a problem.)

The sideline content overlay will word-wrap to two virtual lines.

without diff-hl:
<img width="1496" alt="SCR-20230407-ugmf" src="https://user-images.githubusercontent.com/1028971/230635649-da817169-c6a5-4ed9-b679-e8a8cbb6009a.png">

with diff-hl:
<img width="1496" alt="image" src="https://user-images.githubusercontent.com/1028971/230635928-c47e6a0f-08c3-48ca-bdd7-e3688ce82191.png">

**Comment by jcs090218** [comment]
Hmm.. I am not able to reproduce this. Can you to provide more information? Maybe try `emacs -q`.

**Comment by ginqi7** [comment]
<img width="1479" alt="image" src="https://user-images.githubusercontent.com/1028971/230699855-c6dcc1cf-0ac2-4d2f-99f2-fe52153696c0.png">

I try to use minimal configuration like:

``` emacs-lisp
(require 'diff-hl)
(require 'sideline)
(require 'sideline-eldoc)
(global-diff-hl-mode)
(setq sideline-backends-skip-current-line t ; don't display on current line
      sideline-order-left 'down             ; or 'up
      sideline-order-right 'up              ; or 'down
      sideline-format-left "%s   "          ; format for left aligment
      sideline-format-right "%s"     ; format for right aligment
      sideline-priority 100             ; overlays' priority
      sideline-display-backend-name nil)
(add-hook 'emacs-lisp-mode-hook #'sideline-mode)
(defun my-backend-async (command)
  "Example async backend."
  (cl-case command
    (`candidates
     (cons :async
           (lambda (callback &rest _)
             (funcall callback '("info 1" "info 2" "info 3")))))))
(setq sideline-backends-right '(my-backend-async)) ; use `sideline-backends-right' for right alignment
(setq sideline-backends-left nil) ; use `sideline-backends-right' for right alignment
```

My environment is :
1. MacOS M1
2. emacs 30
3. sideline and diff-hl use latest commit

sideline: 2fa3e671
diff-hl: 67ce270

**Comment by jcs090218** [comment]
Thanks for the minimal configuration! However, I'm still not able to reproduce this bug... I'm on Windows with Emacs `29.0.60`. Maybe something new with Emacs 30? 🤔 Can you try other Emacs versions to see if it occurs?

![2023-04-08 10 52 53](https://user-images.githubusercontent.com/8685505/230735942-f9c15fb8-8dfc-4b89-a459-af0b91d5d220.png)

**Comment by ginqi7** [comment]
It needs that diff-hl-mode fringe and sideline-mode overlay are displayed on a same empty line.

So you could try to create a file in a git project, and commit it, and modify a empty line.

make sure a diff-hl-mode fringe shows on left and sideline overlay display on right. 

<img width="1496" alt="image" src="https://user-images.githubusercontent.com/1028971/230750782-246fa28b-2e23-46af-9f4e-f8eeb700e967.png">

**Comment by jcs090218** [comment]
Thanks! It's fixed in #11.

**Comment by ginqi7** [comment]
Thank you for your quick response. It's fixed.
